src/python/genny/genny_auto_tasks.py

```diff
@@ -0,0 +1,95 @@
+"""
+Output json that can be used as input to evergreen's generate.tasks, representing genny workloads to be run
+"""
+import argparse
+import os
+import re
+import sys
+import subprocess
+import textwrap
+
+from shrub.config import Configuration
+from shrub.command import CommandDefinition
+from shrub.variant import TaskSpec
+
+def to_snake_case(str):
+	"""
+	Converts str to snake_case, useful for generating test id's
+	From: https://stackoverflow.com/questions/1175208/elegant-python-function-to-convert-camelcase-to-snake-case
+	:return: snake_case version of str.
+	"""
+	s1 = re.sub('(.)([A-Z][a-z]+)', r'\1_\2', str)
+	return re.sub('([a-z0-9])([A-Z])', r'\1_\2', s1).lower()
+
+def modified_workload_files():
+	"""
+	Returns a list of filenames for workloads that have been modified according to git, relative to origin/master.
+	:return: a list of filenames in the format subdirectory/Task.yml
+	"""
+	# Return the names of files in ../workloads/ that have been added or modified since the common ancestor of HEAD and origin/master
+	try:
+		out = subprocess.check_output('git diff --name-only --diff-filter=AM $(git merge-base HEAD origin/master) -- ../workloads/', shell=True)
```

Comment:
The git filter is excellent and bulletproof!

Reply:
Is the same true for modules as well? Since this script is considering the git history in `genny` that was set with set-module. I think it might be helpful for me to discuss offline.

Reply:
I ran the whole file through PyCharm's reformat functionality, so that should be set.
When I took out the `origin/master` bit the git command had no output so I think it may be necessary.